### PR TITLE
Add support for tab collections

### PR DESCRIPTION
**Pull request by Ev-1**
Add support tab collections. Allows opening collections of pickers separately. For example having a separate collection for git pickers.

I tried to use tabs for indenting since that is what has been used for the most part. I also haven't written much lua, so I don't know what is considered best practices.

I also tried keeping the interface to the "tabs" mostly unchanged, which means it is unaware of the collections. Making the rest of the code base aware of this change could allow removing some code. 

Also, I really like the plugin :) . Allows me to use a lot of the pickers I otherwise wouldn't because it would require too many binds.

### CODE REVIEW

**Comment by FabianWirth** [comment]
the PR is awesome! thanks a lot :rocket: 
i checked it out and it works like a charm. I have little experience with lua as well but the code looks great to me!

I'd keep it the way it is now, without the codebase knowing a lot about collections.. at least until i got into testing nvim plugins a bit more. I always feel like hell could break loose by any change without having proper test :D But as soon as I added some tests, refactoring a bit more sounds like a good idea..

File: README.md [hunk 1]
```diff
@@ -78,23 +81,48 @@ require("search").setup({
...
+      }
+    }
+  }
+})
+``` 
+
 ### known issues
 - pickers with more-than-average loading time (like lsp related, or http sending pickers) can feel a bit off, since the UI will wait for them to be ready.
 - heavily custom configured telescope settings (like in many nvim distros) might lead to unexpected errors, please open an issue if you encounter any.
+- A window with no available pickers can cause neovim to hang.
```
Comment (FabianWirth) [inline README.md:125]: could you provide a minimal config for this? in my case - if I add a collection / default list with only broken telescope pickers I get a stack overflow.. is that the issue or is it something different?

File: lua/search/tabs/collection.lua [hunk 2]
```diff
@@ -9,7 +9,7 @@ function TabCollection:new(opts)
 		local tab = Tab:new(t, id)
 		tab_list[id] = tab
 	end
-	local id = opts.initial_id or 1
```
Comment (FabianWirth) [inline lua/search/tabs/collection.lua]: yea in the next refactor i'll unify naming of those xD

File: README.md (hunk 1 above)
Comment (Ev-1) [reply README.md:125]: It is a stack overflow at least. What I did was add a collection with only git pickers, with the available func from the readme
```lua
      local is_git_dir = function ()
        return vim.fn.isdirectory(".git") == 1
      end
      
      -- other config
        collections = {
          git = {
            initial_tab = 1, -- Git files
            tabs = {
              { name = "Git files", tele_func = telescope.git_files, available = is_git_dir },
              { name = "Branches", tele_func = telescope.git_branches, available = is_git_dir },
              { name = "Commits", tele_func = telescope.git_commits, available = is_git_dir },
              { name = "Status", tele_func = telescope.git_status, available = is_git_dir },
              { name = "Stash", tele_func = telescope.git_stash, available = is_git_dir },
            }
          },
        }

```
If I'm not in a git directory the stack will overflow.

Comment (FabianWirth) [reply README.md:125]: ah okay I, i'll add this as an issue :)

**Pull request merged**